etl/src/v2/pipeline.rs

```diff
@@ -11,28 +11,26 @@ use crate::v2::schema::cache::SchemaCache;
 use crate::v2::state::store::base::{StateStore, StateStoreError};
 use crate::v2::state::table::TableReplicationPhase;
 use crate::v2::workers::apply::{ApplyWorker, ApplyWorkerError, ApplyWorkerHandle};
-use crate::v2::workers::base::{Worker, WorkerHandle, WorkerWaitError, WorkerWaitErrors};
+use crate::v2::workers::background::{
+    BackgroundWorker, BackgroundWorkerError, BackgroundWorkerHandle,
+};
+use crate::v2::workers::base::{Worker, WorkerHandle, WorkerWaitErrors};
 use crate::v2::workers::pool::TableSyncWorkerPool;
 
 #[derive(Debug, Error)]
 pub enum PipelineError {
-    #[error(
-        "Both the apply worker and table sync workers failed. Apply worker error: {0}, Table sync workers errors: {1}"
-    )]
-    BothWorkerTypesFailed(WorkerWaitError, WorkerWaitErrors),
-
-    #[error("The apply worker failed: {0}")]
-    ApplyWorkerFailed(#[from] WorkerWaitError),
-
-    #[error("Table sync workers failed: {0}")]
-    TableSyncWorkersFailed(WorkerWaitErrors),
```

Comment:
Simplified error handling for now, it will be completely revamped in the future.